blockchain/reactor.go

```diff
@@ -308,6 +314,13 @@ FOR_LOOP:
 					// still need to clean up the rest.
 					bcR.Switch.StopPeerForError(peer, fmt.Errorf("BlockchainReactor validation error: %v", err))
 				}
+				peerID2 := bcR.pool.RedoRequest(second.Height)
+				peer2 := bcR.Switch.Peers().Get(peerID2)
+				if peer2 != nil && peer2 != peer {
+					// NOTE: we've already removed the peer's request, but we
+					// still need to clean up the rest.
+					bcR.Switch.StopPeerForError(peer2, fmt.Errorf("BlockchainReactor validation error: %v", err))
```

Comment:
In case VerifyCommit fails, it seems that we don't know which block is wrong. Does it make sense stopping both peers for error since we are not sure which one is wrong?

Reply:
currently stopping both peers is ok, then it'll fetch block from other peers.

Reply:
I have open a new issue to think about this #2622